src/dlgConnectionProfiles.cpp

```diff
@@ -1778,140 +1765,128 @@ void dlgConnectionProfiles::slot_connectToServer()
-            notificationAreaIconLabelError->show();
-            notificationAreaMessageBox->setText(QString("%1\n%2").arg(notificationAreaMessageBox->text(), tr("Port number must be above zero and below 65535.\r\n")));
-            port_entry->setPalette(mErrorPalette);
-            validPort = false;
-            valid = false;
+        if ((num >= 65536) || (num <= 0) || !ok) {
+            addNotification(tr("Port number must be above zero and below 65535."));
+            port_entry->setStyleSheet(mStyleSheetError);
+            validProfile = false;
+        } else {
+            removeNotification(tr("Port number must be above zero and below 65535."));
         }
 
+        removeNotification(tr("Mudlet is not configured for secure connections."));
+        port_ssl_tsl->setStyleSheet("");
 #if defined(QT_NO_SSL)
         port_ssl_tsl->setEnabled(false);
         port_ssl_tsl->setToolTip(tr("Mudlet is not configured for secure connections."));
         if (port_ssl_tsl->isChecked()) {
-            notificationAreaIconLabelError->show();
-            notificationAreaMessageBox->setText(QString("%1\n%2").arg(notificationAreaMessageBox->text(), tr("Mudlet is not configured for secure connections.\n\n")));
+            addNotification(tr("Mudlet is not configured for secure connections."));
             port_ssl_tsl->setEnabled(true);
-            validPort = false;
-            valid = false;
+            port_ssl_tsl->setStyleSheet(mStyleSheetError);
+            validProfile = false;
         }
 #else
+
+        removeNotification(tr("Mudlet can not load support for secure connections."));
         if (!QSslSocket::supportsSsl()) {
             if (port_ssl_tsl->isChecked()) {
-                notificationAreaIconLabelError->show();
-                notificationAreaMessageBox->setText(QString("%1\n%2").arg(notificationAreaMessageBox->text(), tr("Mudlet can not load support for secure connections.\n\n")));
-                validPort = false;
-                valid = false;
+                addNotification(tr("Mudlet can not load support for secure connections."));
+                port_ssl_tsl->setStyleSheet(mStyleSheetError);
+                validProfile = false;
             }
         } else {
             port_ssl_tsl->setEnabled(true);
             port_ssl_tsl->setToolTip("");
         }
 #endif
+
         QUrl check;
         QString url = host_name_entry->text().trimmed();
         check.setHost(url);
         if (!check.isValid()) {
-            notificationAreaIconLabelError->show();
-            notificationAreaMessageBox->setText(QString("%1\n%2").arg(notificationAreaMessageBox->text(), tr("Please enter the URL or IP address of the Game server.\n\n%1").arg(check.errorString())));
-            host_name_entry->setPalette(mErrorPalette);
-            validUrl = false;
-            valid = false;
+            addNotification(tr("Please enter the URL of the Game server.%1").arg(check.errorString()));
```

Comment:
If you are not including a line-feed before `check.errorString()` in the message then you will need a space after the `.` and before the `%1` replacement.